moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/action_based_controller_handle.h

```diff
@@ -69,13 +69,30 @@ class ActionBasedControllerHandle : public ActionBasedControllerHandleBase
 {
 public:
   ActionBasedControllerHandle(const std::string& name, const std::string& ns)
-    : ActionBasedControllerHandleBase(name), namespace_(ns), done_(true)
+    : ActionBasedControllerHandleBase(name), namespace_(ns), done_(true), nh_("~")
   {
     controller_action_client_.reset(new actionlib::SimpleActionClient<T>(getActionName(), true));
     unsigned int attempts = 0;
-    while (ros::ok() && !controller_action_client_->waitForServer(ros::Duration(5.0)) && ++attempts < 3)
-      ROS_INFO_STREAM_NAMED("moveit_simple_controller_manager", "Waiting for " << getActionName() << " to come up");
+    double timeout;
+    nh_.param("trajectory_execution/controller_connection_timeout", timeout, 5.0);
 
+    if (timeout == 0.0)
+    {
+      ROS_WARN_STREAM_NAMED("moveit_simple_controller_manager", "No controller specified. Waiting indefinitely...");
+      while (ros::ok() && !controller_action_client_->waitForServer(ros::Duration(1.0)))
+      {
+        ROS_ERROR_STREAM_NAMED("moveit_simple_controller_manager", "Waiting for " << getActionName() << " to come up");
+        ros::Duration(1).sleep();
+      }
+    }
+    else
+    {
+      while (ros::ok() && !controller_action_client_->waitForServer(ros::Duration(timeout)) && ++attempts < 3)
+      {
+        ROS_ERROR_STREAM_NAMED("moveit_simple_controller_manager", "Waiting for " << getActionName() << " to come up");
+        ros::Duration(1).sleep();
+      }
```

Comment:
The logic here is a bit weird, isn't it?
If I see this correctly, the new "timeout" parameter means the controller manager will wait up to 3*(timeout+1) seconds in total for the controller because of the loop and the additional sleep.
Could you remove the `sleep`, call `waitForServer` with timeout/3 seconds, and change the default timeout to 15 seconds instead?
That would give the parameter an intuitive meaning and pretty much retain previous behavior.

I'm sorry for not having the time to look at that during WMD.  :(

Reply:
You are fully right. We just added the sleep to avoid having the warning message being printed all the time. In this way we have it printed only once per second. Is there a better way of doing this?

No problem.. We were already running against time.

Reply:
You are fully right. We just added the sleep to avoid having the warning message being printed all the time. In this way we have it printed only once per second. Is there a better way of doing this?

No problem.. We were already running against time.